### ISSUE TITLE
[JAVA]need a configuration to disable java fluent setter

### ISSUE THREAD

**Issue by mumutu** [swagger-api/swagger-codegen#5099]
Also want a configuration to diable java fluent setter.
 Here is my problem:
 I have a Schema Object which has a property like settlementNo then the generated class will have two
setter

```
 public ConditionRequest settlementNo(String settlementNo) {
    this.settlementNo = settlementNo;
    return this;
  }
  public void setSettlementNo(String settlementNo) {
    this.settlementNo = settlementNo
```

then with the default configuration of `spring-fox` and `spring-boot` , in the `spring-fox-swagger-ui`  the `ConditionRequest`  would have two property 
```
settlementNo(the right one)

tlementNo(the wrong one)

```
And if both of the property has the value, `tlementNo` will overwrite `settlementNo`

**Comment by cbornet** [comment]
This looks like a springfox issue to me. Springfox should only consider javabean setters.

**Comment by mumutu** [comment]
@cbornet first I thougth it was the springfox isuue, 
then I use only `spring-boot` with default configuration 

here is the controller
```
@RestController
public class MyTestController {

    @PostMapping("/hello")
    public String world(@RequestBody ConditionRequest request){

        return request.getSettlementNo();
    }
}

```

using postman with payload

```
{
  "tlementNo": "world"
}

```

then return the `world`   I don't think it is the expected result

**Comment by mumutu** [comment]
I think it would be nice if all the fluent setter  will be given a `JsonIgnore` annotations to avoid potential problem. or there is a configuration to disable the fluent setter code~~

**Comment by cbornet** [comment]
So this would be a Jackson weird behavior ! In that case, I guess `JsonIgnore` would be the best solution. Does it solve your issue ?

**Comment by mumutu** [comment]
@cbornet yes~ I using my own pojo.mustache to solve it .... so should I keep this issue open ？

**Comment by cbornet** [comment]
Yes, keep it open because it seems we need to add JsonIgnore on the fluent setters.
You could just change the title.

**Comment by dreameral** [comment]
> pojo

Hi, I also want to customize pojo.mustache. But do I need to override all the .mustache files? When I specified the mustaches directory in pom.xml, it threw an error stating that model.mustache is missing.
